drake/systems/plants/RigidBodySystem.h

```diff
@@ -237,6 +295,51 @@ namespace Drake {
   public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
   };
+
+  /** RigidBodySensor
+   * @brief interface class for elements which define a sensor which reads the state of a rigid body system
+   */
+  class DRAKERBSYSTEM_EXPORT RigidBodySensor {
+  public:
+    RigidBodySensor(RigidBodySystem* sys, const std::string& name) : sys(sys), name(name) {}
+    virtual ~RigidBodySensor() {}
+
+    virtual size_t getNumOutputs() const { return 0; }
+    virtual Eigen::VectorXd output(const double& t, const KinematicsCache<double>& rigid_body_state) const = 0;
+
+  protected:
+    RigidBodySystem* sys;
```

Comment:
you're right.  that's not ideal.  it's coming from the fact that the parser is a method of RigidBodySystem, so this pointer gets generated from this.  I don't want to get into enable_shared_from_this here, I don't think.  Is it cleaner to turn this into a reference?  Even a const reference?